src/vivliostyle/table.js

```diff
@@ -1225,19 +1350,43 @@ goog.scope(function() {
     };
 
     /**
+     * @override
+     */
+    function adjustCellHeight(cellFragment, formattingContext) {
+        var repetitiveElements = formattingContext.getRepetitiveElements();
+        if (!repetitiveElements) return;
+
+        var vertical = formattingContext.vertical;
+        var column = cellFragment.pseudoColumn.column;
+        var cellContentElement = column.element;
+        var cellElement = cellFragment.cellNodeContext.viewNode;
+
+        var cellElementRect = column.clientLayout.getElementClientRect(cellElement);
+        if (vertical) {
+            var width = (cellElementRect.right - column.footnoteEdge - repetitiveElements.calculateElementHeight());
+            adapt.base.setCSSProperty(cellContentElement, "width", width + "px");
+        } else {
+            var height = (column.footnoteEdge - repetitiveElements.calculateElementHeight() - cellElementRect.top);
```

Comment:
2, 3 and 4 should have been addressed by acb4f6374e627efe5367ca89aa12fe2ccaf28054.